### ISSUE TITLE
As a local lead, trainer manager, admin I can export filtered results as CSV

### ISSUE THREAD

**Issue by dupreesi** [yalla-coop/connect5#564]
- this is the same functionality as currently on the app but with applied filters

**Comment by RamyAlshurafa** [comment]
@dupreesi 
we have one button in the user/session results page, should we make export button for each section(behavioural/feedback/train trainers feedback)?

**Comment by dupreesi** [comment]
@RamyAlshurafa there are 2 options and they both related to filtering: 
1) We have 1 filter button on top of the results page -> applies to all sections simultaneously -> this means 1 export CSV button 

2) We have filter and export buttons for each section... 

I would say that 1) is a bit simpler and totally sufficient... 

What do you think?

**Comment by RamyAlshurafa** [comment]
@dupreesi 
so, what data to be exported? the responses? behavioral? feedback?

**Comment by dupreesi** [comment]
@RamyAlshurafa thanks for thinking through...

Maybe better, if possible, to have multiple exports:
1) Responses - top level as it is now -> gets all responses 

2) Individual exports for each section -> gets all responses for relevant categories (not aggregated (no formulae etc. but just raw responses for relevant questions)

Filtering would get only those responses relevant (e.g. get me female responses for session types 1 and 3 related to local lead xyz...) 

So for exporting no aggregation of results... would this be possible?

**Comment by RamyAlshurafa** [comment]
@dupreesi 
yeah I agree, but for the individual section, I would recommend to export it as aggregated data and after passing it to the formulae. the same shape of data as we have for charts
what do you think?

**Comment by dupreesi** [comment]
@RamyAlshurafa would you be able to set up one example row of how this data would look like in a spreadsheet? Thanks a lot!

**Comment by RamyAlshurafa** [comment]
ah sorry @dupreesi 

I'm not familiar with CSV,I was thinking to make the question at the top header of the file(with big cell) then make nested headers for each survey.  but after some searching, I think it not allowed to make merged cells
so this won't work in CSV but can be atchevied in `xcel`,  should we make that in xcel? or just export the raw responses as it's in CSV? what do you think

**Comment by dupreesi** [comment]
hey @RamyAlshurafa I think your idea is great to merge cells but for simplicity lets output raw responses i'd say. people can always use spreadsheets to aggregate data etc... 
would that be ok?

**Comment by RamyAlshurafa** [comment]
yeah sounds good.
